### PR TITLE
feat: Update supported Python and Django versions

### PR DESCRIPTION
**Pull request by adamantike**
Minimal Django version is now 4.2 (oldest supported version upstream [1]) Minimal Python version is now 3.8 (oldest Python version supported by Django 4.2)

[1] https://www.djangoproject.com/download/#supported-versions

### CODE REVIEW

**Review by carltongibson — COMMENTED** [review]
Looks great. Just the one comment.

File: tox.ini [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [tox]
 envlist =
-       {py37,py38,py39,py310}-django32,
-       {py38,py39}-{django40,django41,django42},
-       {py310, py311}-{django41,django42,latest},
-       {py310, py311, py312}-{django41,django42,django50,latest},
+       {py38,py39,py310}-django32,
```
[the remark this answers is not in the document]
Comment (adamantike) [reply tox.ini]: Good catch! I also found some combinations were duplicated (e.g. Python 3.11 / Django 4.2), so moved it to a line per Django version.

**Review by carltongibson — APPROVED** [review]
Perfect.

**Comment by carltongibson** [comment]
Thanks @adamantike — I'll roll a release over the weekend.

**Comment by carltongibson** [comment]
Turned out that was _this afternoon_ https://pypi.org/project/django-filter/24.1/

**Comment by adamantike** [comment]
Thank you for the quick release! I can confirm the unneeded `django.test` import time reported in #1641 is now gone.

image

**Comment by carltongibson** [comment]
Lovely. Thanks for the report and PRs!

**Pull request merged**